routes.go

```diff
@@ -302,6 +304,41 @@ func (s *Server) publishEvent(w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+// publishEvent gets cloud native events and converts it to cloud event and publishes to a transport to send
+//it to the consumer
+func (s *Server) publishHwEvent(w http.ResponseWriter, r *http.Request) {
+	defer r.Body.Close()
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		respondWithError(w, err.Error())
+		return
+	}
+	hwEvent := v1hwevent.CloudNativeEvent()
+	if err = json.Unmarshal(bodyBytes, &hwEvent); err != nil {
+		respondWithError(w, err.Error())
+		return
+	} // check if publisher is found
+	pub, err := s.pubSubAPI.GetPublisher(hwEvent.ID)
+	if err != nil {
+		localmetrics.UpdateEventPublishedCount(hwEvent.ID, localmetrics.FAIL, 1)
+		respondWithError(w, fmt.Sprintf("no publisher data for id %s found to publish event for", hwEvent.ID))
+		return
+	}
+	ceEvent, err := hwEvent.NewCloudEvent(&pub)
+	if err != nil {
+		localmetrics.UpdateEventPublishedCount(pub.Resource, localmetrics.FAIL, 1)
+		respondWithError(w, err.Error())
+	} else {
+		s.dataOut <- &channel.DataChan{
+			Type:    channel.EVENT,
+			Data:    ceEvent,
+			Address: pub.GetResource(),
+		}
+		localmetrics.UpdateEventPublishedCount(pub.Resource, localmetrics.SUCCESS, 1)
```

Comment:
where are local metrics changes? pls run make test  and make lint

Reply:
localmetrics.go

Reply:
make test and make lint all passed. I don't see the need to change localmetrics, unless we want to make the metrics name more generic: `cne_events_api_published` => `cloud_events_api_published`?

https://github.com/redhat-cne/rest-api/blob/main/pkg/localmetrics/localmetrics.go#L41

Reply:
make test and make lint all passed. I don't see the need to change localmetrics, unless we want to make the metrics name more generic: `cne_events_api_published` => `cloud_events_api_published`?

https://github.com/redhat-cne/rest-api/blob/main/pkg/localmetrics/localmetrics.go#L41

Reply:
I am lookign for UpdateEventPublishedCount in localmetrics pakcage